src/main/kotlin/me/kject/internal/call/Caller.kt

```diff
@@ -1,18 +1,90 @@
 package me.kject.internal.call
 
-import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.*
+import me.kject.annotation.Inject
+import me.kject.annotation.With
 import me.kject.call.CallBuilder
 import me.kject.dependency.trace.DependencyTraceBuilder
+import me.kject.exception.call.BadParameterException
+import me.kject.exception.call.CallCanceledException
+import me.kject.exception.call.CallFailedException
+import me.kject.exception.call.MultipleWithsException
+import me.kject.internal.KJectImpl
+import me.kject.internal.Registry
+import java.lang.reflect.InvocationTargetException
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.reflect.KClass
 import kotlin.reflect.KFunction
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.callSuspendBy
+import kotlin.reflect.full.findAnnotations
+import kotlin.reflect.full.hasAnnotation
+import kotlin.reflect.jvm.isAccessible
 
 object Caller {
 
     suspend fun <T> call(
         function: KFunction<T>,
         builder: CallBuilder<T>.() -> Unit,
-        dependencyTraceBuilder: DependencyTraceBuilder
-    ) : Deferred<T> {
-        TODO()
+        dependencyTraceBuilder: DependencyTraceBuilder,
+    ): Deferred<T> {
+        val arguments = CallBuilderImpl(function).apply(builder).getParameters()
+
+        val parameterMap = mutableMapOf<KParameter, Any?>()
+        for (parameter in function.parameters) {
+            if (parameter.hasAnnotation<Inject>()) (parameter.type.classifier as? KClass<*>)
+                ?.let { parameterMap[parameter] = Registry[it] ?: Registry.create(it, dependencyTraceBuilder) }
+            else if (parameter in arguments) parameterMap[parameter] = arguments[parameter]
+            else if (!parameter.isOptional && !parameter.type.isMarkedNullable)
+                throw BadParameterException(parameter, function)
+        }
+
+        var tactic: With.Tactic? = null
+        var directTactic = false
+        for (with in function.findAnnotations<With>()) {
+            val value = KJectImpl.getContextValue(with.context)
+
+            if (value == 2) {
+                if (directTactic) throw MultipleWithsException(function)
+
+                tactic = with.tactic
+                directTactic = true
+            }
+
+            if (value == 1 && !directTactic) {
```

Comment:
magic number

Reply:
Was resolved with the implementation of me.kject.internal.KJectImpl.Context